caproto/_state.py

```diff
@@ -257,6 +265,12 @@ def process_command_type(self, role, command_type):
         for chan in self.channels.values():
             chan.states._fire_state_triggered_transitions(role)
 
+    def disconnect(self):
+        self.states = {CLIENT: DISCONNECTED, SERVER: DISCONNECTED}
+        # Notify channels on this circuit.
+        for chan in self.channels.values():
+            chan.states._fire_state_triggered_transitions(CLIENT)
```

Comment:
does the order that this happens in matter?

Touching private state on attributes is a bit of tight coupling (what are they, c++ friends?), would it be better to have a single `update_state` public method on the channels?